lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -397,9 +397,15 @@ sub _prepare_settings {
     my @keys;
     if ($hp->{settings}) {
         for my $k (keys %{$hp->{settings}}) {
-            $k = trim $k;
             my $value = trim $hp->{settings}->{$k};
-            $k =~ s/[^\]\[0-9a-zA-Z_\+]//g;
+            $k = trim $k;
+            my $ktmp = $k;
+            my %invalid;
+            $ktmp =~ s/([^\]\[0-9a-zA-Z_\+])/$invalid{$1} = 1; ''/eg;
```

Comment:
You could save one line:
```suggestion
            my %invalid;
            my $ktmp = $k =~ s/([^\]\[0-9a-zA-Z_\+])/$invalid{$1} = 1; ''/egr;
```

Reply:
No this doesn't work. When doing this, `$ktmp` and `$k` have both the same value afterwards - I just tested it.

Reply:
Edit: When using the `/r` flag, it actually works.